iothub/device/src/NumericHelpers.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Globalization;
+
+namespace Microsoft.Azure.Devices.Client
+{
+    internal class NumericHelpers
+    {
+        internal static bool TryCastNumericTo<T>(object input, out T result)
+        {
+            if (TryGetNumeric(input))
+            {
+                try
+                {
+                    result = (T)Convert.ChangeType(input, typeof(T), CultureInfo.InvariantCulture);
+                    return true;
+                }
+                catch
+                {
+                }
+            }
+
+            result = default;
+            return false;
+        }
+
+        private static bool TryGetNumeric(object expression)
+        {
+            if (expression == null)
```

Comment:
I know we agreed to bypass braces for logging statements, but not for any other cases.